routes/chorals.js

```diff
@@ -72,26 +74,70 @@ router.get('/new', function(req, res, next) {
 router.get('/:choralId', function(req, res, next) {
   var choralId = req.params.choralId;
   var userModel = res.locals.userModel;
-  var tabs = [];
-
-  let getTabs = new Promise((resolve, reject) => {
-    client.hgetall(choralId, (err, data) => {
-      var parsedData = JSON.parse(data.device_data);
-      for( key in parsedData ) {
-        var val = parsedData[key];
-        if( !isNaN(parseFloat(val)) && isFinite(val) ) {
-          tabs.push(key);
-        }
+
+  // 1. get sample rate of the chorals
+  // 2. default currently is 1 hour of data, so get (3600/sampleRate) data points
+  // 3. put them into object where {tabName: [data_points]} so FE doesn't need to do any work
+  let p = new Promise((resolve, reject) => {
+    ret = {};
+    Choral.findOne({ choralId: choralId }, (err, choral) => {
+      if (err) {
+        logHelper.createLog("error", 'Choral does not exist: ' + err, ["chorals", "delete"]);
+        console.log(err);
+        res.flash('error', 'Choral does not exist');
+        return res.send('404'); // notify client of failure
       }
-      resolve(tabs)
+      ret.sampleRate = choral.sampleRate;
+      resolve(ret);
     });
   });
 
-  getTabs.then((tabs) => {
+  p.then((ret) => {
+    return new Promise((resolve,reject) => {
+      client.hgetall(choralId, (err, data) => {
+        var tabs = [];
```

Comment:
I guess we could see an error if sample rate is very long and we try accessing this before there's any data.